### PR TITLE
Get more OTP itinerary info + fix wrong coords order

### PR DESCRIPTION
**Pull request by khamaileon**

### CODE REVIEW

File: routingpy/routers/valhalla.py [hunk 1]
```diff
@@ -282,6 +282,7 @@ def parse_direction_json(response, units):
             factor = 0.621371 if units == "mi" else 1
             distance += int(leg["summary"]["length"] * 1000 * factor)
 
+        geometry = [coord[::-1] for coord in geometry]
```
Comment (nilsnolde) [inline routingpy/routers/valhalla.py:285]: hm, lat/lon is actually the wrong order. `Direction` object exposes lon,lat tuples by definition, which plays much nicer with virtually all geospatial formats & libs except the oddballs leaflet, google maps api and very few more.

File: routingpy/routers/opentripplanner_v2.py [hunk 2]
```diff
@@ -198,6 +203,7 @@ def _parse_legs(self, legs):
             geometry.extend(points)
             distance += int(leg["distance"])
 
+        geometry = [coord[::-1] for coord in geometry]
```
Comment (nilsnolde) [inline routingpy/routers/opentripplanner_v2.py:206]: will this be lng/lat or lat/lng?

File: routingpy/routers/valhalla.py (hunk 1 above)
Comment (nilsnolde) [reply routingpy/routers/valhalla.py:285]: how can this pass the tests though?!

File: routingpy/routers/opentripplanner_v2.py (hunk 2 above)
Comment (khamaileon) [reply routingpy/routers/opentripplanner_v2.py:206]: Yes, it's lon, lat as specified in the Direction class.

File: routingpy/routers/valhalla.py (hunk 1 above)
Comment (khamaileon) [reply routingpy/routers/valhalla.py:285]: I wonder if it wouldn't be a good idea to review how the tests are made. I'd like to see a snapshot system of API responses that can be easily updated. And we could also add live tests in the pipeline if we manage not to have all the API keys and not to make too many requests. What do you think?

Comment (nilsnolde) [reply routingpy/routers/valhalla.py:285]: > My Google routes are displayed, but not Valhalla's.

That'd be surprising.. I didn't work intensely with routingpy in a while, but we're using it extensively in other projects with Valhalla specifically and I'd hope we'd have caught such an obvious error. I'll have a try later myself. In any case, if there's a problem, it's the polyline decoding and it should be fixed there (utils.py).

> And we could also add live tests in the pipeline if we manage to have all the API keys and not to make too many requests

Yeah, that was heavily worked on a while ago by @chrstnbwnkl : https://github.com/gis-ops/routingpy/pull/85. However, there was a problem with ORS IIRC. And then priorities shifted massively and it didn't make it into master yet.

File: routingpy/routers/opentripplanner_v2.py (hunk 2 above)
Comment (nilsnolde) [reply routingpy/routers/opentripplanner_v2.py:206]: Can we fix this 3 lines further up? I.e. `geometry.extend(list(reversed(points))`

File: routingpy/routers/valhalla.py (hunk 1 above)
Comment (khamaileon) [reply routingpy/routers/valhalla.py:285]: @nilsnolde Don't bother looking. After a second look, I think the problem is on my side 😅

File: routingpy/routers/opentripplanner_v2.py (hunk 2 above)
Comment (khamaileon) [reply routingpy/routers/opentripplanner_v2.py]: Done

File: routingpy/routers/valhalla.py (hunk 1 above)
Comment (khamaileon) [reply routingpy/routers/valhalla.py]: In the end it was Google's that were reversed

File: routingpy/routers/google.py [hunk 3]
```diff
@@ -369,12 +369,8 @@ def parse_direction_json(response, alternatives):
                 duration += leg["duration"]["value"]
                 distance += leg["distance"]["value"]
                 for step in leg["steps"]:
-                    geometry.extend(
-                        [
-                            list(reversed(coords))
-                            for coords in utils.decode_polyline5(step["polyline"]["points"])
-                        ]
-                    )
+                    geometry.extend(utils.decode_polyline5(step["polyline"]["points"]))
```
Comment (nilsnolde) [inline routingpy/routers/google.py:372]: yeah good catch. we refactored the polyline stuff (a few years ago IIRC), and this must've slipped.

**Review by nilsnolde — APPROVED** [review]

**Pull request merged**